src/libsyntax/ast.rs

```diff
@@ -454,12 +454,21 @@ pub enum NestedMetaItemKind {
     Literal(Lit),
 }
 
+/// A name of `MetaItem`.
+#[derive(Clone, Eq, RustcEncodable, RustcDecodable, Hash, Debug, PartialEq)]
+pub enum MetaItemName {
+    /// Unscoped name, e.g. `name` in `#[name(..)]`.
+    Single(Name),
+    /// Scoped name, e.g. `rustfmt::skip` in `#[rustfmt::skip]`.
+    Scoped(Vec<Name>),
+}
+
 /// A spanned compile-time attribute item.
 ///
-/// E.g. `#[test]`, `#[derive(..)]` or `#[feature = "foo"]`
+/// E.g. `#[test]`, `#[derive(..)]`, `#[rustfmt::skip]` or `#[feature = "foo"]`
 #[derive(Clone, PartialEq, Eq, RustcEncodable, RustcDecodable, Hash, Debug)]
 pub struct MetaItem {
-    pub name: Name,
+    pub name: MetaItemName,
     pub node: MetaItemKind,
     pub span: Span,
```

Comment:
Contents of `ast::Attribute` and `ast::MetaItem` are the same thing by definition, but in the compiler they diverged significantly in recent months.
An attribute is a "macro invocation" that contains from a `Path` and a `TokenStream`, `MetaItem` should have the same structure and they should share code, ideally.
I'd rather do this refactoring instead of adding temporary hacks like `Vec<Name>`, preferably in a separate PR.

(The attribute/meta-item `Path` goes through usual path resolution procedure, and that's where resolution of the "tool module" `rustfmt` or `clippy` should be added as well, ideally.)